service/controller/resource/deployment/resource.go

```diff
@@ -107,16 +106,16 @@ func getDeploymentByName(list []*v1.Deployment, name string) (*v1.Deployment, er
 	return nil, microerror.Mask(notFoundError)
 }
 
-func isDeploymentModified(a, b *v1.Deployment) bool {
-	aVersion, ok := a.GetAnnotations()[key.VersionBundleVersionAnnotation]
+func isAnnotationModified(a, b *v1.Deployment, annotation string) bool {
+	aVersion, ok := a.GetAnnotations()[annotation]
 	if !ok {
 		return true
 	}
 	if aVersion == "" {
```

Comment:
When `ok` is `false`, `aVersion` will be the zero value which is the empty string for `string` types. I think you can drop the `!ok` condition as a result.

Reply:
I did that somewhere in the commits and then reverted it back to how it was. Assumed we want to have an explicit check for all cases. I like what you described more too though so I'll bring it back.

Reply:
Done!